Dropout.lua

```diff
@@ -14,34 +14,42 @@ function Dropout:__init(p,v1,inplace)
 end
 
 function Dropout:updateOutput(input)
-   if self.inplace then 
-      self.output = input
-   else
-      self.output:resizeAs(input):copy(input)
-   end
-   if self.train then
-      self.noise:resizeAs(input)
-      self.noise:bernoulli(1-self.p)
-      if self.v2 then
-         self.noise:div(1-self.p)
+   if self.p > 0 then
+      if self.inplace then
+          self.output = input
+      else
+          self.output:resizeAs(input):copy(input)
       end
-      self.output:cmul(self.noise)
-   elseif not self.v2 then
-      self.output:mul(1-self.p)
+      if self.train then
+          self.noise:resizeAs(input)
+          self.noise:bernoulli(1-self.p)
+          if self.v2 then
+             self.noise:div(1-self.p)
+          end
+          self.output:cmul(self.noise)
+      elseif not self.v2 then
+          self.output:mul(1-self.p)
+      end
+   else
+      self.output = input
```

Comment:
this does not respect in-place / out-of-place.
Instead, remove the if condition in line 17, remove this else condition and add:
if self.p >= 0 then
... line 18 to 22
end